redaxo/src/core/lib/util/socket/socket.php

```diff
@@ -39,6 +45,8 @@ class rex_socket
     protected $headers = [];
     /** @vat resource */
     protected $stream;
+    /** @var array<array-key, mixed> */
+    protected $options;
```

Comment:
```suggestion
    protected $options = [];
```

Ich würde den Aufruf von `setOptions` nicht zur Pflicht machen (wäre auch BC Break).